wrangler-core/src/main/java/co/cask/directives/parser/ParseExcel.java

```diff
@@ -159,23 +164,36 @@ public void destroy() {
                 }
                 String value = "";
                 switch (cell.getCellTypeEnum()) {
-                  case STRING:
-                    newRow.add(name, cell.getStringCellValue());
-                    value = cell.getStringCellValue();
-                    break;
 
                   case NUMERIC:
                     if (HSSFDateUtil.isCellDateFormatted(cell)) {
-                      newRow.add(name, cell.getDateCellValue().toString());
+                      if(excelDataType) {
+                        newRow.add(name, cell.getDateCellValue());
+                      }else{
+                        newRow.add(name, cell.getDateCellValue().toString());
+                      }
                       value = cell.getDateCellValue().toString();
                     } else {
-                      newRow.add(name, cell.getNumericCellValue());
+                      if(excelDataType) {
+                        newRow.add(name, cell.getNumericCellValue());
+                      }else{
+                        newRow.add(name, Double.toString(cell.getNumericCellValue()));
```

Comment:
Shouldnt we rather use here - `newRow.add(name, String.valueOf(cell.getNumericCellValue()))`

Reply:
both sataement doing the same thing. Double class have its own to string method thats why it is preferable and commonly use .